src/Xamarin.Android.Build.Tasks/Tasks/ResolveAssemblies.cs

```diff
@@ -201,17 +206,42 @@ void AddAssemblyReferences (DirectoryAssemblyResolver resolver, ICollection<stri
 			}
 
 			LogMessage ("{0}Adding assembly reference for {1}, recursively...", new string (' ', indent), assembly.Name);
+			resolutionPath.Add (assembly.Name.Name);
 			indent += 2;
+
 			// Add this assembly
 			if (!topLevel && assemblies.All (a => new AssemblyNameDefinition (a, null).Name != assembly.Name.Name))
 				assemblies.Add (fullPath);
 
 			// Recurse into each referenced assembly
 			foreach (AssemblyNameReference reference in assembly.MainModule.AssemblyReferences) {
-				var reference_assembly = resolver.Resolve (reference);
-				AddAssemblyReferences (resolver, assemblies, reference_assembly, false);
+				AssemblyDefinition reference_assembly;
+				try {
+					reference_assembly = resolver.Resolve (reference);
+				} catch (FileNotFoundException ex) {
+					var references = new StringBuilder ();
+					for (int i = resolutionPath.Count - 1; i >= 0; i--) {
+						references.Append ('`');
+						references.Append (resolutionPath [i]);
+						references.Append ('`');
+						if (i != 0)
+							references.Append (" > ");
+					}
+
+					string missingAssembly = Path.GetFileNameWithoutExtension (ex.FileName);
+					string message = $"Could not find assembly `{missingAssembly}`, referenced by {references}.";
+					if (MonoAndroidHelper.IsFrameworkAssembly (ex.FileName)) {
+						LogError ("XA9999", $"{message} Perhaps it doesn't exist in the Mono for Android profile?");
```

Comment:
`XA9999` was an *example*. We need to come up with a "proper" value, based on https://github.com/xamarin/java.interop/blob/master/src/Java.Interop.Tools.Diagnostics/Java.Interop.Tools.Diagnostics/Diagnostic.cs.

Perhaps `XA2002` would be relevant ("Can not resolve reference: {0}")? Or `XA2006` ("Reference to metadata item '{0}' (defined in '{1}') from '{2}' could not be resolved")?

Reply:
Ok thanks, I've not gotten to "invent" an error code yet.